src/fam_struct_wrapper.rs

```diff
@@ -0,0 +1,590 @@
+    /// a supplied slice of [`FamStruct::Entry`](trait.FamStruct.html#associatedtype.Entry).
+    ///
+    /// # Arguments
+    ///
+    /// * `entries` - The vector of [`FamStruct::Entry`](trait.FamStruct.html#associatedtype.Entry)
+    ///               entries.
+    ///
+    pub fn from_entries(entries: &[T::Entry]) -> FamStructWrapper<T> {
+        let mut adapter = FamStructWrapper::<T>::new(entries.len());
+
+        {
+            let wrapper_entries = adapter.as_mut_fam_struct().as_mut_slice();
+            wrapper_entries.copy_from_slice(entries);
+        }
+
+        adapter
+    }
+
+    /// Get a reference to the actual [`FamStruct`](trait.FamStruct.html) instance.
+    ///
+    pub fn as_fam_struct_ref(&self) -> &T {
+        &self.mem_allocator[0]
+    }
+
+    /// Get a mut reference to the actual [`FamStruct`](trait.FamStruct.html) instance.
+    ///
+    pub fn as_mut_fam_struct(&mut self) -> &mut T {
+        &mut self.mem_allocator[0]
+    }
+
+    /// Get a pointer to the [`FamStruct`](trait.FamStruct.html) instance.
+    ///
+    pub fn as_fam_struct_ptr(&self) -> *const T {
+        self.as_fam_struct_ref()
+    }
+
+    /// Get a mutable pointer to the [`FamStruct`](trait.FamStruct.html) instance.
+    ///
+    pub fn as_mut_fam_struct_ptr(&mut self) -> *mut T {
+        self.as_mut_fam_struct()
+    }
+
+    /// Get the elements slice.
+    ///
+    pub fn as_slice(&self) -> &[T::Entry] {
+        self.as_fam_struct_ref().as_slice()
+    }
+
+    /// Get the mutable elements slice.
+    ///
+    pub fn as_mut_slice(&mut self) -> &mut [T::Entry] {
+        self.as_mut_fam_struct().as_mut_slice()
+    }
+
+    /// Reserves capacity for at least `additional` more
+    /// [`FamStruct::Entry`](trait.FamStruct.html#associatedtype.Entry) elements.
+    /// If the capacity is already reserved, this method doesn't do anything.
+    ///
+    fn reserve(&mut self, additional: usize) {
+        let desired_capacity = self.len + additional;
+        if desired_capacity <= self.capacity {
+            return;
+        }
+
+        let current_mem_allocator_len = self.mem_allocator.len();
+        let required_mem_allocator_len = FamStructWrapper::<T>::mem_allocator_len(desired_capacity);
+        let additional_mem_allocator_len = required_mem_allocator_len - current_mem_allocator_len;
+
+        self.mem_allocator.reserve(additional_mem_allocator_len);
+        self.capacity = FamStructWrapper::<T>::fam_len(self.mem_allocator.capacity());
+    }
+
+    /// Updates the length of `self` to the specified value.
+    /// Also updates the length of the `T` structure and of `self.mem_allocator` accordingly.
+    /// Reserves additional capacity if the len is increased.
+    /// Deallocates unnecessary memory if the len is decreased.
+    ///
+    /// # Error: When len is greater than the max possible len it returns Error::SizeLimitExceeded
+    ///
+    fn set_len(&mut self, len: usize) -> Result<(), Error> {
```

Comment:
It seems that when `set_len` causes the length to increase, the newly added elements are based on basically random memory contents (I don't think reserving capacity also initializes the memory to a particular value). If that's true, is this a part of the `set_len` method contract, or we should try to zero-initialize the new elements like we do for `new`?